### PR TITLE
Remove device get binding from more sensor samples

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Continued incremental progress to using DT-aware device getters throughout the samples.

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
@gmarull since you do a lot of sensor review and have expressed a preference for DEVICE_DT_GET_ONE, I've tried to use it here, along with accompanying documentation changes to hopefully help users debug.

**Review by alexanderwachter — APPROVED** [review]

File: samples/sensor/ams_iAQcore/src/main.c [hunk 1]
```diff
@@ -11,12 +11,13 @@
...
 {
-	const struct device *dev;
+	const struct device *dev = DEVICE_DT_GET_ONE(ams_iaqcore);
 	struct sensor_value co2, voc;
 
-	dev = device_get_binding(DT_LABEL(DT_INST(0, ams_iaqcore)));
-	if (!dev) {
-		printk("Failed to get device binding");
+	if (!device_is_ready(dev)) {
+		printf("Device %s is not ready; "
```
Comment (jfischer-no) [inline samples/sensor/ams_iAQcore/src/main.c:18]: print**k**("Device %s is not ready; "

**Comment by gmarull** [comment]
> @gmarull since you do a lot of sensor review and have expressed a preference for DEVICE_DT_GET_ONE, I've tried to use it here, along with accompanying documentation changes to hopefully help users debug.

I actually have no strong preference. What I prefer is consistency, I think it is confusing for a newcomer to find both `DEVICE_DT_GET_ONE` and `DEVICE_DT_GET_ANY`. I'd prefer to settle on one choice for in-tree samples.

**Comment by avisconti** [comment]
DEVICE_DT_GET_ONE() works by compatible strings, which is good as far as there is only one instance for a particular device in the DT, which seems to be the case for all this sample. But I have done a lot of testing for multi-instance (I don't recall if I have ever commited those samples, probably they are only local to my repo). So, my feeling is that this PR is good, but DEVICE_DT_GET_ONE()  working with compatibles is not suitable for more complex cases. Not sure if I expressed my doubts in a clear way...

**Pull request closed without merging**